init/eessi_environment_variables

```diff
@@ -38,6 +38,45 @@ if [ -d $EESSI_PREFIX ]; then
           break
         fi
       done
+
+      # we need to make sure that errexit shell option (set -e) is not enabled,
+      # since archdetect will produce non-zero exit code if no accelerator was found
+      if [[ "$-" =~ e ]]; then
+        errexit_shell_option_set='yes'
+        set +e
+      else
+        errexit_shell_option_set='no'
+      fi
+
+      # to be able to grab exit code of archdetect trying to detect accelerators,
+      # we can not run it via $(...), so we have to redirect the output to a temporary file
+      tmpout=$(mktemp)
+      ${EESSI_INIT_DIR_PATH}/eessi_archdetect.sh accelpath 2>&1 > $tmpout
+      accelpath_exit_code=$?
+
+      if [[ "$errexit_shell_option_set" == "yes" ]]; then
+        set -e
+      fi
+
+      if [[ $accelpath_exit_code -eq 0 ]]; then
+          export EESSI_ACCEL_SUBDIR=$(tail -1 $tmpout && rm -f $tmpout)
```

Comment:
Not sure why you bother creating a `$tmpout` since it is only used here?

Reply:
Ah, maybe it is redirection stuff I don't fully understand

Reply:
The issue is that I couldn't do this:

```
EESSI_ACCEL_SUBDIR=$(${EESSI_INIT_DIR_PATH}/eessi_archdetect.sh accelpath)
```

because then I couldn't seem to get the exit code correctly, I assumed since it's run in a subshell (although that doesn't actually make much sense to me, and testing with dummy commands doesn't confirm this)